src/cli/one_helper/oneprovider_helper.rb

```diff
@@ -14,203 +14,418 @@
+            return if stdin.empty?
+
+            content = stdin
         end
+
+        begin
+            JSON.parse(content, :symbolize_names => true)
+        rescue JSON::ParserError => e
+            source = file ? "file: #{file}" : 'stdin'
+            STDERR.puts "Invalid JSON in #{source} - #{e.message}"
+            exit(-1)
+        end
+    end
+
+    def get_user_values(user_inputs)
+        return {} if user_inputs.nil? || user_inputs.empty?
+
+        ask_user_inputs(user_inputs)
     end
 
-    # Returns provider pool
-    def factory_pool(_)
-        OneProvision::ProviderPool.new(@client)
+    def ask_user_inputs(inputs)
+        puts 'There are some parameters that require user input.'
+
+        answers = {}
+
+        inputs.each do |input|
+            name        = input[:name]
+            description = input[:description] || ''
+            type        = input[:type]
+            default     = input[:default]
+            match       = input[:match]
+
+            puts "  * (#{name}) #{description} [type: #{type}]"
+            header = '    '
+            header += "Press enter for default (#{default}). " if default
+
+            answer = nil
+
+            type = case type
+                   when /\Amap\(/ then 'map'
+                   when /\Alist\(/ then 'list'
+                   else type
+                   end
+
+            case type
+            when 'string'
+                if match&.dig(:type) == 'list'
+                    options = match[:values] || []
+                    options.each_with_index {|opt, i| puts "    #{i}: #{opt}" }
+                    puts
+
+                    loop do
+                        print "#{header}Please type the selection number: "
+                        raw = STDIN.readline.strip
+
+                        if raw.empty?
+                            answer = default
+                            break if options.include?(answer)
+                        else
+                            index = raw.to_i rescue nil
+                            answer = options[index] if index && index >= 0
+                            break if answer
+                        end
+
+                        puts '    Invalid selection, please try again.'
+                    end
+                else
+                    print header
+
+                    loop do
+                        answer = STDIN.readline.strip || ''
+                        answer = OpenNebulaHelper.editor_input if answer == '<<EDITOR>>'
+                        answer = default if answer.to_s.empty?
+
+                        break unless answer.to_s.empty?
+
+                        print '    Input cannot be empty. Please try again: '
+                    end
+                end
+            when 'number'
+                min = match&.dig(:values, :min)
+                max = match&.dig(:values, :max)
+
+                begin
+                    range_msg = min && max ? " (#{min} to #{max})" : ''
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

A cross-site scripting (XSS) vulnerability was detected. User-controlled data is being output without proper sanitization, allowing an attacker to inject malicious scripts. This could lead to session hijacking, malware installation, or phishing attacks. Always sanitize and encode user inputs before including them in the output, using context-appropriate encoding methods. Learn More - https://cwe.mitre.org/data/definitions/79.html.